### PR TITLE
SOUTH BRONX PARADISE BABY!!! (New Contraband Candy Bar)

### PR DESCRIPTION
**Pull request by ArcaneMusic**
image

## About The Pull Request

Adds a new contraband piece of candy, the South Bronx Paradise bar. It's the new hotness! It's for dieting! [Kinda!](https://www.youtube.com/watch?v=5GJGDw1mEjc) It's tasty for lizards, flies, all the races that like eating literal larva candy bars. 

It is also exceptionally disgusting, and eats away at the amount of of nutrition you have. You should probably only eat one of these if you **REALLY** want to lose weight.
## Why It's Good For The Game

We have a real and true content drought here, and that drought, is content from the bronx. Also, this will act as a good snack food for most of the non-humans on station, who will get a mood boost from the food types this has.

## Changelog
:cl:
add: Adds a new weight loss bar to the contraband on snack vending machines.
/:cl:

### CODE REVIEW

**Comment by trollbreeder** [comment]
makes lipoplasty, lipolicide and lipid extracters redundant for species that like gross food
powercrêpe

**Comment by ArcaneMusic** [comment]
> makes lipoplasty, lipolicide and lipid extracters redundant for species that like gross food
> powercrêpe

I hear ya, the lipolicide doesn't do much in the amounts of a single candy bar. I switched around the chemical values a bit so that it's mostly organic slurry that's causing the "weight loss" element. Also, keeping in mind that they're more limited within the vending machine, and knowing how exceptionally weak the nutrient loss from vomiting is, you're more likely to run out of money than actually lose much weight with several of these.

**Comment by trollbreeder** [comment]
> -snip- knowing how exceptionally weak the nutrient loss from vomiting is, you're more likely to run out of money than **actually lose much weight with several of these.**

> You should probably only eat **one** of these if you **REALLY** want to lose weight.

PR description implies it's way stronger than you imply in your comment. This causes confusion for me and others. Eeeee.

**Comment by Farquaar** [comment]
I understand it's a reference, but I think a gross candy-bar without weight-loss effects would be a better item.

File: tgstation.dme [hunk 1]
```diff
@@ -2855,7 +2855,6 @@
 #include "code\modules\surgery\plastic_surgery.dm"
 #include "code\modules\surgery\prosthetic_replacement.dm"
 #include "code\modules\surgery\remove_embedded_object.dm"
-#include "code\modules\surgery\stomachpump.dm"
```
Comment (itseasytosee) [inline tgstation.dme]: Why

**Review by itseasytosee — COMMENTED** [review]
I know it would be a lot more work but I don't think liposide is the right choice for a chemical. Mabey something that increased the amount of weight loss from each step? for me this just seems like an easy out for anyone who eats to mutch and now has to run about for a while or go to a doctor/chemist

**Comment by Ryll-Ryll** [comment]
Oughta add an admin only version with the correct "South Bronx Parasite" label that implants you with an alien embryo to really be true to the source

File: tgstation.dme (hunk 1 above)
Comment (trollbreeder) [reply tgstation.dme]: how in the actual fuck did this happen

**Comment by tralezab** [comment]
Just makes me miss south bronx glockroach way more

File: tgstation.dme (hunk 1 above)
Comment (ArcaneMusic) [reply tgstation.dme]: More like how did that happen when I already readded the file 3 TIMES NOW.
I'll fix it when I get back

**Comment by carshalash** [comment]
Finally I can roleplay as Carl Brutananadilewski

**Comment by GunHog** [comment]
Upgrade your meme!

It should continue to slowly drain the host until his nutrition level reaches 0. At that point, It has a small chance per tick to make the host vomit out his liver. Thus the host must continue to eat, install a nutriment implant, or cure the infection.

Suggestions for a cure:
- Removal surgery (if coded as a parasite)
- Any type of acid injection or injection (if coded as a disease). RP wise, you are drinking acid to strengthen the host's stomach acids to be strong enough to kill the larva.

File: code/modules/food_and_drinks/food/snacks_vend.dm [hunk 2]
```diff
@@ -13,6 +13,18 @@
 	tastes = list("candy" = 1)
 	foodtype = JUNKFOOD | SUGAR
 
+/obj/item/reagent_containers/food/snacks/candy/bronx
+	name = "south bronx paradise"
+	desc = "Lose weight, guaranteed! Wait, the package actally says 'south bronx parasite'..."
```
Comment (ATH1909) [inline code/modules/food_and_drinks/food/snacks_vend.dm]: "actally"

should you capitalize references to the name of this candy bar? is it a brand or just a type of dish?

**Comment by ArcaneMusic** [comment]
Alright, with feedback I've now made the South Bronx Paradise bar into an actual infectious disease. Eating it will give the eater a slow acting but still notable disease that will steadily cause the eater to lose weight. The disease is cured via the loss of the liver, surgical or otherwise. While it is noticeable, there are clear warnings about eating one, and you'll notice that something is wrong for sure before any of the more serious symptoms begin to affect you.
Additionally, as the most dangerous result (Losing your liver) can be notably harmful, I've limited the number of candy bars in the vendor to 1 each.
image

**Comment by Tlaltecuhtli** [comment]
is the disease spreadable? like feeding a monky and then sampling the blood into the round start donuts

**Comment by ArcaneMusic** [comment]
> 
> 
> is the disease spreadable? like feeding a monky and then sampling the blood into the round start donuts

Not at all, it's no-spread just like appendicitis.

File: code/datums/diseases/parasitic_infection.dm [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+				affected_mob.emote("cough")
+		if(2)
+			if(prob(10))
+				if(prob(50))
+					to_chat(affected_mob, "<span class='notice'>You feel the weight loss already!</span>")
+				affected_mob.adjust_nutrition(-3)
+		if(3)
+			if(prob(20))
+				to_chat(affected_mob, "<span class='notice'>You're... REALLY starting to feel the weight loss.</span>")
+				affected_mob.adjust_nutrition(-6)
```
Comment (XDTM) [inline code/datums/diseases/parasitic_infection.dm]: the chat messages are going to be quite spammy at that chance to proc

Comment (ArcaneMusic) [reply code/datums/diseases/parasitic_infection.dm]: Gotchu fam

File: code/modules/food_and_drinks/food/snacks_vend.dm (hunk 2 above)
Comment (ArcaneMusic) [reply code/modules/food_and_drinks/food/snacks_vend.dm]: Brand name, I... I guess? Should be fine now.

File: code/modules/food_and_drinks/food/snacks_vend.dm [hunk 4]
```diff
@@ -13,6 +13,32 @@
 	tastes = list("candy" = 1)
 	foodtype = JUNKFOOD | SUGAR
 
+/obj/item/reagent_containers/food/snacks/candy/bronx
+	name = "South Bronx Paradise bar"
+	desc = "Lose weight, guaranteed! Wait, the package actually says 'South Bronx Parasite'..."
```
Comment (actioninja) [inline code/modules/food_and_drinks/food/snacks_vend.dm]: if you're doing the whole "change after one examine" thing, don't mention the parasite part in the desc

File: code/modules/food_and_drinks/food/snacks_vend.dm [hunk 5]
```diff
@@ -13,6 +13,32 @@
...
+	if(ishuman(eater))
+		var/mob/living/carbon/human/carl = eater
+		var/datum/disease/P = new /datum/disease/parasite()
+		carl.ForceContractDisease(P, FALSE, TRUE)
+
+/obj/item/reagent_containers/food/snacks/candy/bronx/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Geeze, you need to get to get your eyes checked. You should look again...</span>")
+	desc = "Lose weight, guaranteed! Caramel Mocha Flavor! WARNING: PRODUCT NOT FIT FOR HUMAN CONSUMPTION. CONTAINS LIVE DIAMPHIDIA SPECIMENS."
+	name = "South Bronx Parasite bar"
```
Comment (actioninja) [inline code/modules/food_and_drinks/food/snacks_vend.dm]: have some kind of flag for this so it's not being spammed/updating name every time

File: code/datums/diseases/parasitic_infection.dm [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	permeability_mod = 1
+	desc = "If left untreated the subject will passively lose nutrients, and eventually lose their liver."
+	severity = DISEASE_SEVERITY_HARMFUL
+	disease_flags = CAN_CARRY|CAN_RESIST
+	spread_flags = DISEASE_SPREAD_NON_CONTAGIOUS
+	required_organs = list(/obj/item/organ/liver)
+	bypasses_immunity = TRUE
+
+/datum/disease/parasite/stage_act()
+	..()
```
Comment (actioninja) [inline code/datums/diseases/parasitic_infection.dm]:
```suggestion
	. = ..()
```

**Review by actioninja — CHANGES_REQUESTED** [review]

**Pull request merged**